### ISSUE TITLE
Question about iCloud Sync

### ISSUE THREAD

**Issue by rogerluan** [2FAuth/2FAuth-iOS#2]
Hey 👋 

I've been using this great iOS app for over a year now, and although I know that Apple supports native 2FA now, I still have good use for it. 

I'd like to know how can I sync my 2FA between multiple devices, using iCloud Sync. I downloaded the app on a 2nd device, which's signed in using the same Apple ID, but the 2FA codes weren't populated automatically. Then, on the 2nd device, I'm afraid that if I toggle iCloud Sync and set a new password (as it requests), it could override my shared iCloud storage with an empty database 😬 

So, first of all - is it possible? Cuz I know there's iCloud Sync doesn't sync _across_ devices, but only on the same device (for backup and device upgrade purposes) 😅 

Looking forward to continue using this app! Appreciate any input 🙏

**Comment by podkovyrin** [comment]
Sorry for the late answer.
> I downloaded the app on a 2nd device, which's signed in using the same Apple ID, but the 2FA codes weren't populated automatically. 

That's correct. iCloud is turned off by default until you provide it with a passphrase to encrypt/decrypt your data.
It syncs across devices using the same Apple ID if you manually turn on the sync and input the same passphrase. If the input passphrase is unable to decrypt your data you will be asked to try again or delete iCloud records to start over.

**Comment by rogerluan** [comment]
@podkovyrin thanks for getting back to me. 
The UX is so weird during this flow that even after reading what you're saying I'm unsure if my data will be safe. On the 2nd device (macOS), after downloading the app, the iCloud Backup is off by default. Upon attempting to turn it own, it shows this:

initial screen|screen asking me to "set up" passphrase, which sounds like it'd be a NEW passphrase?|screen to enter passphrase for the first time|then the passphrase is asked again (to confirm), strongly indicating a NEW one is being created
--|--|--|--
<img width="210" src="https://user-images.githubusercontent.com/8419048/172140817-6435ca40-873b-4d6a-8529-ae25bd7145c7.png">|<img width="210" src="https://user-images.githubusercontent.com/8419048/172140832-f921ac2e-5f6c-4bbd-846f-e87b16aa508c.png"> |<img width="210" src="https://user-images.githubusercontent.com/8419048/172140951-bd7a711b-19df-4026-b850-18eea718d434.png"> |<img width="210" src="https://user-images.githubusercontent.com/8419048/172140873-1f6f0ab8-556f-4d9a-83f8-b7c960851abb.png"> 

<sub>(You can click on the screenshots to enlarge them)</sub>

Can you see how this is confusing? It totally looks like a new one is being created instead of an existing one being "checked/verified" 😬 I'm afraid that I turn this on on the 2nd device → a new passphrase is created → it overrides my iCloud Sync backup → and since the 2nd device is empty, it will sync things over to the cloud, and sync it back into the 1st device (iOS), which will then erase all my 2FA 😱

**Comment by podkovyrin** [comment]
Agree. That should have been much user-friendlier.
Though, there's no plan to improve this since Apple's own 2FA support via iCloud Keychain seems pretty decent and already has built-in Safari support.

**Comment by rogerluan** [comment]
But ultimately my question is, if I use the same passphrase created on Device 1, and enter it twice on Device 2, everything will be sync'd (and there's a zero chance of things being reset)? @podkovyrin

**Comment by podkovyrin** [comment]
99% chance that it's gonna be fine. If you have services that are not restorable or the restoration process is too difficult it's always better to backup their 2FA credentials to any other app just not to worry.

**Comment by podkovyrin** [comment]
Also, there's a trick: turn on airplane mode on your first device, restore your backup on the second device and if everything is ok safely turn off airplane mode back again. If anything goes wrong (again, I doubt it would) you will still have a local copy of 2FA records on your first device until a push from CloudKit comes in.

**Comment by rogerluan** [comment]
Ah, that's a great tip! Thank you so much! 😊 I'll try it!

**Comment by rogerluan** [comment]
It worked! 🚀 Thanks!

**Comment by rogerluan** [comment]
Closing this issue now :) I still hope this UX can be improved at some point, though. Something like:

If we can detect somehow that the user has something in their 2FAuth iCloud, then redirect the user to a "confirm your passphrase" flow, that explicitly says "hey we identified you have enabled 2FAuth in another device, enter the same passphrase in this device to sync it..." and if they enter the wrong passphrase we'd have ideally the option to either:
a. Wipe the whole thing from the Device A and iCloud, starting from a clean slate (destructive action)
b. Configure a 2nd cloud-backed 2FAuth storage (Device A and Device B are intentionally out of sync and have different passphrases, and would never be merged together)
c. Try again with another passphrase to try to match the same passphrase as Device A.

Hope this can be implemented some day 😊 thanks for the amazing simple and secure 2FAuth app. :)
